lib/vnet/user_process_windows.go

```diff
@@ -75,3 +98,66 @@ func runPlatformUserProcess(ctx context.Context, config *UserProcessConfig) (pm
 	})
 	return pm, nil
 }
+
+// secureCredDir sets ACLs so that the current user can write credentials files
+// to dir but cannot read them. Only the LocalSystem account is allowed to read
+// and delete the files.
+func secureCredDir(dir string) error {
+	u, err := user.Current()
+	if err != nil {
+		return trace.Wrap(err, "getting current OS user")
+	}
+	// Uid is documented to be the user's SID on Windows.
+	userSID := u.Uid
+	// S-1-5-18 is the well-known SID (security identifier) for the LocalSystem
+	// service account, which the VNet windows service runs as.
+	const serviceSID = "S-1-5-18"
+	// O: Set Owner as current user (Windows won't allow LocalSystem to be the owner)
+	// G: Set Primary Group as current user
+	// D: Discretionary ACL
+	//   Grant GA (Generic All) to LocalSystam
+	//   Grant GW (Generic Write) to the current user so we can add files
+	//   Deny GR (Generic Read) for the current user so other processes can't read the credentials
+	//   OICI (Object Inherit, Container Inherit) propagates permissions to files/folders
```

Comment:
I appreciate the comment block above, this would be extremely opaque without it!